### PR TITLE
LOG-1195: Add Openshift Ingest Plugin for ES

### PR DESCRIPTION
**Pull request by lukas-vlcek**
### Description
Configure Elasticsearch to install [Openshift Ingest Plugin](https://github.com/ViaQ/elasticsearch-openshift-ingest-plugin).

/cc @ewolinetz 
/assign @jcantrill 

### Links

- JIRA: <https://issues.redhat.com/browse/LOG-1195>

### CODE REVIEW

**Review by jcantrill — COMMENTED** [review]
/hold

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (jcantrill) [inline elasticsearch/fetch-artifacts-koji.yaml:4]: This will break cpaas because it wont be found

Comment (lukas-vlcek) [reply elasticsearch/fetch-artifacts-koji.yaml:4]: I will comment it out for now... but once the PNC build it setup and running I can put it back, right?

**Comment by lukas-vlcek** [comment]
@jcantrill @ewolinetz I added default pipeline definition and initialization on ES startup. Also I defined the default pipeline in appropriate index templates. Please have a look again.

File: elasticsearch/ci-env.sh
Comment (ewolinetz) [inline elasticsearch/ci-env.sh:7]: s/INGESTI_PLUGIN_URL/INGEST_PLUGIN_URL/

File: elasticsearch/init/0100-seed-pipelines
Comment (ewolinetz) [inline elasticsearch/init/0100-seed-pipelines:27]: we shouldn't need this do we? we aren't doing any `sed` operations...

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (ewolinetz) [inline elasticsearch/fetch-artifacts-koji.yaml:4]: this is commented out because we don't yet have this being built in PNC, correct?

**Comment by ewolinetz** [comment]
> Q: Please let me know if I need to add the new folder `ingest_pipelines` on this line here:

we shouldn't need to because we aren't updating the file at all. no need to do writes there.

File: elasticsearch/init/0100-seed-pipelines
Comment (lukas-vlcek) [reply elasticsearch/init/0100-seed-pipelines:27]: We do not. I will remove it and we can add later if needed.

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (lukas-vlcek) [reply elasticsearch/fetch-artifacts-koji.yaml:4]: corrent

File: elasticsearch/ci-env.sh
Comment (lukas-vlcek) [reply elasticsearch/ci-env.sh]: fixed

File: elasticsearch/Dockerfile
Comment (jcantrill) [inline elasticsearch/Dockerfile:34]: add ```- nvr: org.elasticsearch.plugin.ingest-openshift-ingest-plugin-6.8.1.0_redhat_00001-1``` to ```https://github.com/openshift/origin-aggregated-logging/blob/master/elasticsearch/fetch-artifacts-koji.yaml```

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (jcantrill) [inline elasticsearch/fetch-artifacts-koji.yaml:4]: published version ```- nvr: org.elasticsearch.plugin.ingest-openshift-ingest-plugin-6.8.1.0_redhat_00001-1 ```

File: elasticsearch/ci-env.sh
Comment (jcantrill) [inline elasticsearch/ci-env.sh:7]: If we never intend to pubblish this to an official maven repo then this line should simply default to  use the URL manipulation that you are doing for CI and/or pulling it straight from the Viaq repo

**Review by jcantrill — CHANGES_REQUESTED** [review]

**Comment by ewolinetz** [comment]
@lukas-vlcek can we also fix the `run.sh` file to properly pick up the log4j2.properties file?

File: elasticsearch/ci-env.sh
Comment (lukas-vlcek) [reply elasticsearch/ci-env.sh:7]: This plugin is not generalized enough atm. It makes not sense to push it into any public mvn repo.

**Comment by lukas-vlcek** [comment]
@ewolinetz I think we can (though I would prefer separate PR).

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (lukas-vlcek) [inline elasticsearch/fetch-artifacts-koji.yaml:4]: @jcantrill When this line can be uncommented?

File: elasticsearch/Dockerfile.in
Comment (jcantrill) [inline elasticsearch/Dockerfile.in:61]: 6.8.1.0-redhat-00002

ref: https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=1638435

File: elasticsearch/fetch-artifacts-koji.yaml
Comment (jcantrill) [inline elasticsearch/fetch-artifacts-koji.yaml:4]: org.elasticsearch.plugin.ingest-openshift-ingest-plugin-6.8.1.0_redhat_00002-1

https://brewweb.engineering.redhat.com/brew/buildinfo?buildID=1638435

File: elasticsearch/ci-env.sh
Comment (jcantrill) [inline elasticsearch/ci-env.sh:17]: Shouldnt we "correct" the upstream url to point to a release published here https://github.com/ViaQ/elasticsearch-openshift-ingest-plugin

File: elasticsearch/ci-env.sh
Comment (jcantrill) [reply elasticsearch/ci-env.sh:7]: correct.  that is exactly my point.  we never intend to publish to maven so this change is incorrect (e.g. $MAVEN_REPO_URL)

File: elasticsearch/ci-env.sh
Comment (lukas-vlcek) [reply elasticsearch/ci-env.sh:17]: Ok, I see what you mean @jcantrill
The thing is that we need to decide and control who, how and when cuts new releases for https://github.com/ViaQ/elasticsearch-openshift-ingest-plugin/

At this point it is just manual process. I created some tags in my clone for which I manually built releases and uploaded them to https://github.com/lukas-vlcek/elasticsearch-openshift-ingest-plugin/releases
(^^ This is what @vimalk78 used for his tests btw...)

Now, there is no process that is syncing tags between my clone and official ViaQ upstream.
We can either create tags manually or we can adopt similar approach that is used in prometheus-plugin – which means that whenever there is a commit that includes this property file without `-SNAPSHOT` suffix here ...
https://github.com/vvanholl/elasticsearch-prometheus-exporter/blob/master/gradle.properties#L3

... then a new tag is created automatically and a new release is built and uploaded automatically as well. See https://github.com/vvanholl/elasticsearch-prometheus-exporter/blob/master/.github/workflows/ci.yml#L39

We should decide how we want to control this. I would prefer fully automated approach (i.e. the samy way the prometheus plugin does it), but I am OK if you decide to do it differently.

//cc @ewolinetz

Comment (lukas-vlcek) [reply elasticsearch/ci-env.sh:17]: BTW, I do not think I have privs to create tags on ViaQ repo (the upstream). Which means if we decide we will create and upload release manually then I believe we need to ask @ewolinetz to do this.

**Pull request merged**
